File: hm2/ts/TextBLock.ts
class TextBlock {
    private lines: string[];

    constructor(lines: string[]) {this.lines = lines}

    getLineLength(index: number): number {
        if (index >= 0 && index < this.lines.length) {
            return this.lines[index].length;
        }
        throw new Error("Index out of bounds");
    }

    get vowelCount(): number {
        const vowels = "aeiouаеєиіїоуюяAEIOUАЕЄИІЇОУЮЯ";
        return this.lines.join("").split("").filter(char => vowels.includes(char)).length;
    }
}

File: hm2/cs/TextBlock.cs
using System;
using System.Linq;

class TextBlock
{
    private string[] lines;

    public TextBlock(string[] lines)
    {
        this.lines = lines;
    }

    public int this[int index]
    {
        get
        {
            if (index >= 0 && index < lines.Length)
                return lines[index].Length;
            throw new IndexOutOfRangeException("Індекс за межами масиву.");
        }
    }

    public int VowelCount
    {
        get
        {
            string allText = string.Join("", lines);
            string vowels = "aeiouаеєиіїоуюяAEIOUАЕЄИІЇОУЮЯ";
            return allText.Count(c => vowels.Contains(c));
        }
    }
}